### PR TITLE
Fix abort handling

### PR DESCRIPTION
**Pull request by timursevimli**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md
- [ ] update .d.ts typings

### CODE REVIEW

File: lib/collector.js [hunk 1]
```diff
@@ -28,10 +28,10 @@ class Collector {
     if (typeof timeout === 'number' && timeout > 0) {
       this.#timeout = AbortSignal.timeout(timeout);
       this.#signal = AbortSignal.any([this.#signal, this.#timeout]);
-      this.#timeout.addEventListener('abort', () => {
+      this.#signal.addEventListener('abort', () => {
         if (Object.keys(this.defaults).length > 0) this.#default();
         if (this.done) return;
-        this.fail(this.#timeout.reason);
+        this.fail(this.#signal.reason);
```
Comment (axbuglak) [inline lib/collector.js:34]:
```suggestion
        this.fail(reason);
```

File: lib/collector.js [hunk 2]
```diff
@@ -28,10 +28,10 @@ class Collector {
     if (typeof timeout === 'number' && timeout > 0) {
       this.#timeout = AbortSignal.timeout(timeout);
       this.#signal = AbortSignal.any([this.#signal, this.#timeout]);
-      this.#timeout.addEventListener('abort', () => {
+      this.#signal.addEventListener('abort', () => {
         if (Object.keys(this.defaults).length > 0) this.#default();
         if (this.done) return;
```
Comment (axbuglak) [inline lib/collector.js:33]:
```suggestion
        const reason = this.#signal.reason
        if (reason.endsWith('timeout') && Object.keys(this.defaults).length > 0) this.#default();
        if (this.done) return;
```
After this fix, default values will be used after all abort signals. So we should check if this is a signal from timer.

**Review by axbuglak — COMMENTED** [review]
After this fix, default values will be used after all abort signals. So we should check if this is a signal from timer. I think it might look something like that. And we already have one test for timeout, so maybe it would be better to just add `dc.signal.addEventListener('abort'` to it?

**Comment by timursevimli** [comment]
> After this fix, default values will be used after all abort signals. So we should check if this is a signal from timer. I think it might look something like that. And we already have one test for timeout, so maybe it would be better to just add `dc.signal.addEventListener('abort'` to it?

Thanks for the suggestions! Indeed, a timeout test already exists, but I missed it. I think we can follow your suggestion by adding ```dc.signal.addEventListener('abort', (reason) => { ... })``` to the existing test instead of two almost identical tests with timeouts.

I'm a little skeptical about using EndWith in this case. If a comparison is needed there, I think it makes sense to get the error name via ```dc.signal.reason.name```, although I don't think this is necessary to check.

I'm not sure if we should handle the ```this.#default()``` method differently. What I mean is that if the error is caused by a timeout, call this method, and if the error occurs due to some other reason, then do not call it. I believe this is not correct behavior.

@tshemsedinov What do you think about it? Should the handling of the ```this.#default()``` method be different depending on the reasons for the error?

File: lib/collector.js (hunk 2 above)
Comment (timursevimli) [reply lib/collector.js:33]: I'm a little skeptical about using ```.endsWith``` in this case. If a comparison is needed there, I think it makes sense to get the error name via ```dc.signal.reason.name``` and compare like ```dc.signal.reason.name === 'TimeoutError'```, although I don't think this is necessary to check.

**Review by tshemsedinov — APPROVED** [review]

**Pull request merged**